api/v1alpha4/awsmachine_webhook.go

```diff
@@ -49,6 +49,10 @@ var (
 func (r *AWSMachine) ValidateCreate() error {
 	var allErrs field.ErrorList
 
+	if r.Spec.InstanceType == "" {
+		allErrs = append(allErrs, field.Required(field.NewPath("spec.instancetype"), "instancetype is required"))
```

Comment:
`InstanceType` is an optional field in AWSMachine with json tag of `omitempty` so I don't think we can error out when instancetype is empty.
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/api/v1alpha4/awsmachine_types.go#L75

If we want to make it a required field, we need to change the API. @randomvariable, thoughts?

Reply:
This being optional and omitempty was a mistake I think, since the reconciler just ended up passing "" to ec2.RunInstance and failing, so I think it's reasonable to fix.

Ideally, we should mark the package as validation required and  then use the optional tags properly. I think in this case, add `// +kubebuilder:validation:Required` to the field and use OpenAPI validation instead of doing it in the webhook.

Reply:
Agreed that it's good to fix. Just wanted to point out that changing a field from optional to required is a breaking API change and we need to inform the users. 
Using kubebuilder validation marker sounds like a good idea.